src/server/api/routers/applications/applicationsRouter.ts

```diff
@@ -87,34 +87,54 @@ export const applicationsRouter = createTRPCRouter({
           ["id"],
         ),
       ]);
-      const approvedIds = approved.map((a) => a.refUID);
 
-      const { status, ...filter } = input;
-      let ids;
-      if (status === "approved") ids = approvedIds.length ? approvedIds : [""]; // empty string otherwise will fetch all
-      if (status === "pending")
-        ids = applicationsCount // non-approved applications
-          .filter((a) => !approvedIds.includes(a.id))
+      const applicationdsWithStatus = await Promise.all(
+        applicationIds.map(async (a) => ({
+          id: a.id,
+          status: await getApplicationStatus({
+            round: ctx.round,
+            projectId: a.id,
+          }),
+        })),
+      );
+      const { ...filter } = input;
+
+      let ids: string[] = [];
+      if (filter.status === "approved") {
+        ids = applicationdsWithStatus
+          .filter((a) => a.status.status === "approved")
+          .map((a) => a.id);
+      } else if (filter.status === "pending") {
+        ids = applicationdsWithStatus
+          .filter((a) => a.status.status === "pending")
           .map((a) => a.id);
+      } else if (filter.status === "all") {
+        ids = applicationIds.map((a) => a.id);
+      }
```

Comment:
Not sure I understand why this is needed to fetch the `getApplicationStatus` for each attestation? If Line 63 contains the revoked ones we already know which are approved and rejected?

Running `getApplicationStatus` for every application is gonna take a long time, especially with hundreds or thousands of applications.